src/main/java/com/redhat/digkins/DiggerClient.java

```diff
@@ -0,0 +1,56 @@
+package com.redhat.digkins;
+
+import com.offbytwo.jenkins.JenkinsServer;
+import com.redhat.digkins.services.CreateJobService;
+import com.redhat.digkins.util.JenkinsAuth;
+import com.redhat.digkins.util.DiggerClientException;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * Digger Java Client
+ * <p>
+ * Interact with digger jenkins api!
+ */
+public class DiggerClient {
+
+  private final JenkinsServer jenkins;
+
+  public DiggerClient(JenkinsAuth auth) throws URISyntaxException {
+    this.jenkins = new JenkinsServer(new URI(auth.getUrl()), auth.getUser(), auth.getPassword());
```

Comment:
+1 on creating the `java.net.URI` here instead of later. I was gonna suggest that yesterday :)